### PR TITLE
fix description of flag --service-account-key-file

### PR DESCRIPTION
**Pull request by pandaamanda**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Since the constraint between --service-account-key-file and --service-account-signing-key-file is removed in #70967, fix the description of flag --service-account-key-file.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubeapiserver/options/authentication.go [hunk 1]
```diff
@@ -310,8 +310,7 @@ func (o *BuiltInAuthenticationOptions) AddFlags(fs *pflag.FlagSet) {
 			"File containing PEM-encoded x509 RSA or ECDSA private or public keys, used to verify "+
 			"ServiceAccount tokens. The specified file can contain multiple keys, and the flag can "+
 			"be specified multiple times with different files. If unspecified, "+
-			"--tls-private-key-file is used. Must be specified when "+
-			"--service-account-signing-key is provided")
```
Comment (liggitt) [inline pkg/kubeapiserver/options/authentication.go:314]: @mikedanese - that's unfortunate... we don't require this to be explicit when `--service-account-signing-key` is set? falling back to `--tls-private-key-file` is asking for trouble on serving cert rotation.

Comment (pandaamanda) [reply pkg/kubeapiserver/options/authentication.go:314]: --service-account-key-file is required by #95896, so should we modify the description to follow the validation.
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/options/authentication.go#L205

Comment (liggitt) [reply pkg/kubeapiserver/options/authentication.go:314]: actually, `--service-account-key-file` is required as of 1.20 (https://github.com/kubernetes/kubernetes/pull/95896)

drop the `If unspecified, --tls-private-key-file is used.` bit

**Comment by liggitt** [comment]
/close

due to inactivity

feel free to reopen once addressing https://github.com/kubernetes/kubernetes/pull/98901/files#r619340857 is addressed

**Pull request closed without merging**